users/views.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth.tokens import default_token_generator
+from rest_framework import status, viewsets
+from rest_framework.decorators import action, api_view
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+
+from users.models import User
+from users.serializers import UserSerializer
+
+
+@api_view(['GET'])
+def api_root(request, format=None):
+    return Response({
+        'users': reverse('users:user_list', request=request, format=format),
+    })
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def create(self, request, *args, **kwargs):
+        response = super().create(request)
+        email = request.data.get('email', '')
+        user = User.objects.get(email=email)
+        if user:
+            token = default_token_generator.make_token(user)
+            activation_link = 'activate/?user_id={}&token={}'.format(
+                    user.id, token)
+            print(activation_link)
+            url = request.build_absolute_uri(activation_link)
+            print(url)
+        else:
+            print('You need to provide a valid email address.')
+        return response
+
+    @action(detail=False, methods=['GET'])
+    def activate(self, request):
+        user_id = request.query_params.get('user_id', '')
+        confirmation_token = request.query_params.get('token', '')
+        try:
+            user = self.get_queryset().get(pk=user_id)
+        except(TypeError, ValueError,  User.DoesNotExist):
+            user = None
+        print(f'activate: user is {user}')
+        print(f'activate: token is {confirmation_token}')
+        if user is None:
+            return Response('User not found',
+                            status=status.HTTP_400_BAD_REQUEST)
+        if not default_token_generator.check_token(user, confirmation_token):
+            return Response('''Token is invalid or expired.
+            Please request another confirmation email by signing in.''',
+                            status=status.HTTP_400_BAD_REQUEST)
+        user.is_active = True
+        user.save()
```

Comment:
Save -> save all information about the object in Django we can use https://docs.djangoproject.com/en/3.2/ref/models/instances/#specifying-which-fields-to-save

Reply:
I can specify the field to update, I'd like to understand why. Django docs mentions 'slight performance benefit'. Is there any other reason?